tests/test_self.py

```diff
@@ -1239,6 +1239,63 @@ def test_recursive(self):
             code=0,
         )
 
+    def test_ignore_recursive(self):
+        self._runtest(
```

Comment:
It's not very why the tests are checking the ignore option here, would it possible to add asserts and tests the output of ``_discover_files`` directly. I know it's probably a big refactor where we have to create a class or function. Maybe add a docstring for each tests "``failing.py`` would exit with error code 2 if it was analyzed by pylint" ?

Reply:
Added comments + separated unittest for creating list of linted files.